4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-systems-navigation-dedicated-2D-navigation-server.md

---
type: entry
section: systems
subsection: navigation
rank: 0
importance: 3
anchor: dedicated-2d-navigation-server
title: Dedicated 2D navigation server
text: |

>   Ever since its introduction in Godot 4.0, users and navigation nodes interfaced <code class="highlight"><span class="enginetype">NavigationServer2D</span></code> for all their 2D-pathfinding needs. But what if I told you that <code class="highlight"><span class="enginetype">NavigationServer2D</span></code> was hiding something?

Reviewer: ```suggestion
  Ever since its introduction in Godot 4.0, users and navigation nodes interfaced with the <code class="highlight"><span class="enginetype">NavigationServer2D</span></code> for all their 2D-pathfinding needs. But what if I told you that <code class="highlight"><span class="enginetype">NavigationServer2D</span></code> was hiding something?
```

  Previously, <code class="highlight"><span class="enginetype">NavigationServer2D</span></code> was effectively a proxy for <code class="highlight"><span class="enginetype">NavigationServer3D</span></code>; it used the 3D navigation logic but with everything constrained to two axes.

>   However, this had an important caveat. Pure 2D games required an export template that had 3D support. That would obviously bump the final export size of their game.

Reviewer: ```suggestion
  However, this had an important caveat. Pure 2D games using navigation required an export template that had 3D support. That would obviously bump the final export size of their game.
```
For clarity IMO

section: systems
subsection: navigation
rank: 0
importance: 3
anchor: dedicated-2d-navigation-server
title: Dedicated 2D navigation server
text: |

>   Ever since it’s introduction in Godot 4.0, users and navigation nodes interfaced `NavigationServer2D` for all their 2D-pathfinding needs. But what if I told you that `NavigationServer2D` feels… _sus_. It’s hiding something.

Reviewer 2: ```suggestion
  Ever since its introduction in Godot 4.0, users and navigation nodes interfaced `NavigationServer2D` for all their 2D-pathfinding needs. But what if I told you that `NavigationServer2D` was hiding something?
```

>   Until 4.5, well, it **was** _sus_. That’s because we finally managed to create a dedicated 2D navigation server.

Reviewer 2: ```suggestion
```
I find the wording of these sentences a bit confusing; it makes it sound like the creation of a dedicated server is what was "sus". Linking the previous sentence with the next sentence flows more naturally (asking the question of hiding something, then elaborating on what it was hiding).

>   Before this release, `NavigationServer2D` was pretty much just a proxy for `NavigationServer3D`. That’s because we figured we could reuse the 3D logic for it, as 2D navigation can ultimately be represented in 3D by constraining everything in two axis.

Reviewer 2: ```suggestion
  Previously, `NavigationServer2D` was effectively a proxy for `NavigationServer3D`; it used the 3D navigation logic but with everything constrained to two axes.
```

>   Though, this had an important caveat. Namely, pure 2D games required an export template that had 3D support. That would obviously bump the final export size of their game.

Reviewer 2: ```suggestion
  However, this had an important caveat. Pure 2D games required an export template that had 3D support. That would obviously bump the final export size of their game.
```
